MetaDataset.py

```diff
@@ -885,23 +884,106 @@ def init_seq_order(self, epoch=None, seq_list=None):
     """
     assert seq_list is None, "seq_list not supported for %s" % self.__class__
     need_reinit = self.epoch is None or self.epoch != epoch
-    super(CombinedDataset, self).init_seq_order(epoch=epoch, seq_list=seq_list)
+    num_seqs_saved = self._num_seqs
+    super(CombinedDataset, self).init_seq_order(epoch=epoch, seq_list=seq_list)  # resets self._num_seqs
     self.rnd.seed(epoch or 1)
 
     if not need_reinit:
+      self._num_seqs = num_seqs_saved
       return False
 
-    # First init sequence order for sub-datasets as usual to get a list of available sequences. This way sorting and
+    # First init sequence order for sub-datasets as usual to get a list of available sequences. This way, sorting and
     # partition epoch of the individual sub-datasets is still supported. Later we will call init_seq_order again with a
     # sequence list to e.g. apply joint sorting or partition epoch of all sequences.
     for dataset in self.datasets.values():
       dataset.init_seq_order(epoch=epoch)
 
-    self.dataset_sorted_seq_idx_list = []  # We will fill this as we go
-    self.used_num_seqs_per_subset = [0] * len(self.datasets)
+    try:
+      total_num_seqs = sum([self.datasets[k].num_seqs for k in sorted(self.datasets.keys())])
+    except Exception:
+      total_num_seqs = None
+
+    if total_num_seqs is not None:
+      self.used_num_seqs_per_subset = [self.datasets[self.dataset_idx2key_map[dataset_idx]].num_seqs
+                                       for dataset_idx in range(len(self.datasets))]
```

Comment:
The indentation should not be done this way. The indentation should always be 2 spaces (or in certain exceptions maybe 4 spaces). To get that here, just make a line-break after the first `[`.